src/appsignal/cli/diagnose.py

```diff
@@ -0,0 +1,390 @@
+        return self.report["host"]["uid"]["result"]
+
+    def group_id(self) -> str:
+        return self.report["host"]["gid"]["result"]
+
+    def logger_started(self) -> str:
+        if "logger" in self.report and self.report["logger"]["started"]["result"]:
+            return "started"
+        return "not started"
+
+    def working_directory_user_id(self) -> bool:
+        return (
+            "working_directory_stat" in self.report
+            and self.report["working_directory_stat"]["uid"]["result"]
+        )
+
+    def working_directory_group_id(self) -> bool:
+        return (
+            "working_directory_stat" in self.report
+            and self.report["working_directory_stat"]["gid"]["result"]
+        )
+
+    def working_directory_permissions(self) -> bool:
+        return (
+            "working_directory_stat" in self.report
+            and self.report["working_directory_stat"]["mode"]["result"]
+        )
+
+    def lock_path(self) -> str:
+        if "lock_path" in self.report and self.report["lock_path"]["created"]["result"]:
+            return "writable"
+        return "not writable"
+
+
+class DiagnoseCommand(AppsignalCLICommand):
+    @staticmethod
+    def init_parser(parser: ArgumentParser) -> None:
+        parser.add_argument(
+            "--send-report",
+            action="store_true",
+            help="Send the report to AppSignal",
+        )
+        parser.add_argument(
+            "--no-send-report",
+            action="store_true",
+            help="Do not send the report to AppSignal",
+        )
+
+    def run(self) -> int:
+        agent = Agent()
+        agent_json = json.loads(agent.diagnose())
+        self.config = Config()
+        self.send_report = self.args.send_report
+        self.no_send_report = self.args.no_send_report
+        self.agent_report = AgentReport(agent_json)
+
+        self.report = {
+            "agent": {
+                "agent": {
+                    "boot": agent_json["boot"],
+                    "config": {
+                        "valid": agent_json["config"]["valid"],
+                    },
+                    "host": agent_json["host"],
+                    "lock_path": agent_json.get("lock_path"),
+                    "logger": agent_json.get("logger"),
+                    "working_directory_stat": agent_json.get("working_directory_stat"),
+                }
+            },
+            "config": {
+                "options": self.config.options,
+                "sources": self.config.sources,
+            },
+            "host": {
+                "architecture": platform.machine(),
+                "heroku": os.environ.get("DYNO") is not None,
+                "language_version": platform.python_version(),
+                "os": platform.system().lower(),
+                "os_distribution": self._os_distribution(),
+                "root": False,
+                "running_in_container": self.config.option("running_in_container")
```

Comment:
Since the current value is based on what the user has configured, rather than what we determine by reading the host environment, I suggest we omit it from this section. [This section is only meant for what we detect](https://github.com/appsignal/integration-guide/blob/main/features/cli/diagnose.md#host-report), which allows us to compare what we detected against what the user configured. How it's set up now could create confusion by us thinking this is what the package detected.